cmd/metacache-walk.go

```diff
@@ -143,7 +138,16 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 		}
 		dirObjects := make(map[string]struct{})
 		for i, entry := range entries {
+			if len(prefix) > 0 && !strings.HasPrefix(entry, prefix) {
+				// Do do not retain the file, since it doesn't
+				// match the prefix.
+				entries[i] = ""
+				continue
+			}
 			if len(forward) > 0 && entry < forward {
+				// Do do not retain the file, since it doesn't
+				// is lexically smaller than 'forward'
+				entries[i] = ""
```

Comment:
This felt to be right since we do not need to stack this up again in dirStack() as we do not need to go further down if there is no prefix match